Replicator/Worker.hh

```diff
@@ -132,51 +132,50 @@ namespace litecore { namespace repl {
         virtual void afterEvent() override;
         virtual void caughtException(const std::exception& x) override;
 
-        virtual inline std::string formatWithCollection(const char* fmt) const {
-            return format(string(kCollectionLogFormat), collectionID()) + " " + string(fmt);
+        // Add the token for collection info to the format string
```

Comment:
I don't think this will improve things much.  It's not that likely that you will ever log two of the same messages in a row on the same actor, let alone globally.

Reply:
A function that contains a log statement may be called multiple times. I want to work with this case and avoid the search.

Reply:
I don't see how that is going to help.  It will literally only help if you've called the exact same log line twice in a row.

Reply:
I thought this is for the formatting string, not the actual output string. Basically, we have two literals,
A = "{coll#%1}"
B = "this one is per log lexical statement with %i, %d, %*s, etc"
A and B have allocated by the compiler. We want to juxtapose them. The output of the juxtaposition depends on B only. In this function, we cache the result in a vector. 
The log statement can be called multiple times, but juxtapose(A, B) will be the same.

Reply:
A will remain the same, B is variable.  Consider this:

"This is log line 1"
"This is log line 2"
"This is log line 1"

All three of these will result in a lookup, even with your added logic.

The only way for your static caching to have any effect at all is to have the exact same log line called twice in a row with nothing else in between.  Otherwise you just keep setting the new "input" value over and over.  You might get lucky once in a very small while but for 99% it is just added complexity for very little gain.

Reply:
@borrrden you are right, thanks for pointing it out. Main point is that we don't have to search with strings. Comparing pointer may suffice. We may keep an a sorted array of pointers and do binary search.